### DISCUSSION TITLE
Fundamentos do ASP.NET 6 / Introdução e Minimal APIs / Rodando sua primeira App => httpS não aparece.

### DISCUSSION THREAD

**Discussion by weslleyrufino** [balta-io/forum#412] [Tira Dúvidas] [answered]
Falaí Balta, tudo bem?

Ao seguir o passo a passo de criação de um "ASP.NET Core Empty" através do PowerShell, ao executar o dotnet run, não exibe a opção de URL com http**S**. 
![image](https://github.com/balta-io/forum/assets/11825320/e05dc78a-04a0-41b9-834f-93c5a605cc12)

Ao fazer pelo Visual Studio (usei o 2022 especificando a versão .net 6, pois a ultima versão instalada na minha máquina é a 7.0.402) e ao executar, aparece a opção de url com http**S**, acredito que isso ocorra pois na criação do projeto aparece a opção para "Configurar para HTTPS" como pode ver abaixo:

![Microsoft Visual Studio](https://github.com/balta-io/forum/assets/11825320/57fbcedb-e37d-461b-93ce-6697f512450b)

**Tem alguma forma de forçar a geração do httpS através do PowerShell ou teria que fazer alguma outra configuração?**

PS.: Quando criei através do VisualStudio, ao executar deu erro de certificado (ERR_CERT_INVALID), mas importei o mesmo para a pasta "Certificados - Local Computer", "Autoridades de Certificação Raiz Confiáveis" no gerenciador de certificados, feito isso parou o erro. Após fazer isso, mesmo recriando o projeto no PowerShell, o projeto do PowerShell não exibiu o http**S**.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @weslleyrufino, como vai?
Normalmente ele cria HTTP e HTTPS, talvez por não ter encontrado os certificados, ele habilitou apenas o HTTP.

Tenta executar os comandos abaixo para gerar o certificado HTTPS:
```
dotnet dev-certs https --clean
dotnet dev-certs https --trust
```

Depois tenta gerar o projeto novamente pelo comando `dotnet new`

**Reply by weslleyrufino** [reply]
Boa tarde @andrebaltieri , fiz conforme você falou mas sem sucesso. Segue abaixo:
![image](https://github.com/balta-io/forum/assets/11825320/ef434fb4-ce70-431a-b83e-cb0ffcc792e4)

**Comment by andrebaltieri** [comment]
Veja se dentro da pasta Properties, no arquivo launchSettings.json tem alguma configuração com HTTPS, algo assim:
```json
"https": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "https://localhost:7185;http://localhost:5045",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
```

**Reply by weslleyrufino** [reply]
No meu launchSettings.json tem configuração para o HTTPS, segue abaixo: 

{
  "iisSettings": {
    "windowsAuthentication": false,
    "anonymousAuthentication": true,
    "iisExpress": {
      "applicationUrl": "http://localhost:1879",
      "sslPort": 44301
    }
  },
  "profiles": {
    "http": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "http://localhost:5010",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "https": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "https://localhost:7021;http://localhost:5010",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    }
  }
}

**Reply by andrebaltieri** [reply]
Faz uma cópia deste arquivo como backup...
Deixa só a configuração do HTTPS e tenta usar o `dotnet run` por favor.

**Reply by weslleyrufino** [reply]
Falaí Balta, ao remover a configuração http apareceu o https:

![image](https://github.com/balta-io/forum/assets/11825320/26dbfed2-ebf6-40b1-96e4-c409ce71fee8)

Mas o grande problema é que estava sendo criado com a ultima versão do .net que tenho na minha máquina que é a 7.

Forcei o comando abaixo para gerar na versão 6 e aí sim apareceu as opções do http e https:

Comando executado: **dotnet new web --framework net6.0 -n NomeProjeto**

![image](https://github.com/balta-io/forum/assets/11825320/8266acfb-b994-4285-8a87-8e8dc1a64ab3)

**Reply by ddcsilva** [reply]
Opa! Sei que o tópico está antigo, mas segue minha contribuição!

Eu estou usando .NET 8 e tive o mesmo problema. Eu dei uma lida por cima e parece que a partir do .net 7 não vem mais essa configuração automática. Então eu configurei o Kestrel para ele ouvir tanto HTTP quanto HTTPS, dessa forma no program.cs

```
var builder = WebApplication.CreateBuilder(args);

// Configurações do Kestrel para ouvir em HTTP e HTTPS
builder.WebHost.ConfigureKestrel(options =>
{
    options.ListenLocalhost(5273); // Porta HTTP
    options.ListenLocalhost(7232, listenOptions =>
    {
        listenOptions.UseHttps(); // Porta HTTPS
    });
});

var app = builder.Build();

app.MapGet("/", () => "Hello World!");

app.Run();
```

Também, por via das dúvidas, rode o comando no terminal:
`dotnet dev-certs https --trust
`

**Comment by sinvalfelisberto** [comment]
Eu contornei o problema rodando `dotnet run --launch-profile https`

**Reply by luizflorencio** [reply]
Olá pessoal,
usando dotnet 8.0.101 por aqui.

após a dica do sinvalfelisberto eu troquei a ordem dos profiles colocando o https primeiro na lista, assim apenas com o comando `dotnet run` já obtive o as portas para http e https como esperado.

faltou algum ajuste no template de criação do dotnet new web? 

o curioso é que no tutorial da Microsoft o caminho apresentado deveria entregar nos 2 modos... mas na prática hoje não acontece.
https://learn.microsoft.com/pt-br/training/modules/build-web-api-minimal-api/3-exercise-create-minimal-api
